Initial website layout

astrofrog opened the pull request:
This is the initial website layout based on the logo provided by Kyle Barbary and the template pointed out by Eli Bressert. I have created an up-to-date preview here:

http://astrofrog.github.com/astropy-website/index.html

If I make changes in this pull request, I will update that URL accordingly.

Suggestions welcome! Please take a minute to check over all the text. If there are any major additions you would like to make, we can still work on it via pull requests - once this pull request is merged in, we will not publish the website immediately, so there will be time for more work on it if needed.

astrofrog committed:
Initial Sphinx files with bootstrap theme

astrofrog committed:
Updated menubar and css

astrofrog committed:
Added .gitignore

astrofrog committed:
Added menubar template

astrofrog committed:
Removed unused files

astrofrog committed:
Removed relations links from menubar

astrofrog committed:
Added main content and minor tweaks to style

astrofrog committed:
Added line to create _static directory if missing, as suggested by @eteq

astrofrog committed:
Added 'Fork me on GitHub'

astrofrog committed:
Initial version of the team page

astrofrog committed:
Added content for contributing

astrofrog committed:
Added a mention of the vision on the front page, and added the vision

astrofrog committed:
Improved installation instructions

astrofrog committed:
Added logo copyright for Kyle Barbary

astrofrog committed:
Fixed typo

embray commented:
+1!!! Looks gorgeous.
Let me know if you need any help with web-specific stuff. I have a lot of background there. Though you seem well-versed as well.

embray commented:
Hmm...the HTML has some bugs though--doesn't quite validate (they're small issues though).  But these would be problems with the sphinx template.  I assume there's somewhere I can submit patches for it?
</htmlfascist>

astrofrog commented:
@iguananaut - some of the HTML is in our repo, and some is inherited from sphinx. I'll check to see what kind of issues are popping up to see if it's our fault or Sphinx's.

embray commented on _templates/layout.html:
```diff
@@ -0,0 +1,9 @@
+{% extends "!layout.html" %}
+
```
Most of the validation problems with this template seem to stem from the fact that Sphinx is just using the wrong doctype, as the bootstrap theme seems to be HTML5.  I don't know why the theme itself doesn't do this, but inserting this here should fix it I _think_ (not tested):

```
{% block doctype %}
<!DOCTYPE html>
{% endblock %}
```

astrofrog committed:
Fixed missing />

astrofrog commented:
All the errors seem to come from the additions to `layout.html` from the boostrap theme (see `_themes/bootstrap/layout.html`). Does it make more sense to update the DOCTYPE as you are suggesting, or fix the bootstrap template to work with the default DOCTYPE from Sphinx?

embray commented:
I think the theme is fine as it is. It just needs the right doctype.  Two other small tweaks to follow shortly...  Fix the doctype and fix these and it should be all fixed :)

embray commented on _themes/bootstrap/layout.html:
```diff
@@ -0,0 +1,119 @@
+            {% include "menubar.html" %}
+          {% endblock %}
+        </ul>
+        <ul class="nav secondary-nav">
```
&lt;li&gt;

taldcroft commented:
This looks really super and professional.  My only suggestion would be
to get the beautiful logo and "AstroPy" somewhere in the top bar so
that anywhere in the site you always see it.  (E.g. you get straight
to the documentation page from google and it's not immediately obvious
that you are in the AstroPy site apart from the tab title).  Maybe
"Home" should be "@ AstroPy" or "@ AstroPy Home" (where @ is the
logo)?  I'm not entirely convinced that's the right thing, but
hopefully you get the idea.

Also the github image to the right (presumably Fork Me on Github) is
broken in my view.

On Fri, Apr 13, 2012 at 9:54 AM, Thomas Robitaille
reply@reply.github.com
wrote:

> This is the initial website layout based on the logo provided by Kyle Barbary and the template pointed out by Eli Bressert. I have created an up-to-date preview here:
> 
> http://astrofrog.github.com/astropy-website/index.html
> 
> If I make changes in this pull request, I will update that URL accordingly.
> 
> Suggestions welcome! Please take a minute to check over all the text. If there are any major additions you would like to make, we can still work on it via pull requests - once this pull request is merged in, we will not publish the website immediately, so there will be time for more work on it if needed.
> 
> You can merge this Pull Request by running:
> 
>  git pull https://github.com/astrofrog/astropy-website initial-layout
> 
> Or you can view, comment on it, or merge it online at:
> 
>  https://github.com/astropy/astropy-website/pull/1
> 
> -- Commit Summary --
> - Initial Sphinx files with bootstrap theme
> - Updated menubar and css
> - Added .gitignore
> - Added menubar template
> - Removed unused files
> - Removed relations links from menubar
> - Added main content and minor tweaks to style
> - Added line to create _static directory if missing, as suggested by @eteq
> - Added 'Fork me on GitHub'
> - Initial version of the team page
> - Added content for contributing
> - Added a mention of the vision on the front page, and added the vision
> - Improved installation instructions
> 
> -- File Changes --
> 
> A .gitignore (1)
> A Makefile (155)
> A _templates/layout.html (9)
> A _themes/bootstrap/layout.html (119)
> A _themes/bootstrap/menubar.html (6)
> A _themes/bootstrap/relations.html (8)
> A _themes/bootstrap/searchbox.html (7)
> A _themes/bootstrap/sourcelink.html (4)
> A _themes/bootstrap/static/bootstrap-dropdown.js (55)
> A _themes/bootstrap/static/bootstrap-scrollspy.js (107)
> A _themes/bootstrap/static/bootstrap-sphinx.css_t (24)
> A _themes/bootstrap/static/bootstrap.css (2486)
> A _themes/bootstrap/theme.conf (5)
> A astropy.ico (0)
> A conf.py (252)
> A contributing.rst (35)
> A documentation.rst (15)
> A index.rst (45)
> A installation.rst (47)
> A logos/astropy_banner_96.png (0)
> A support.rst (20)
> A team.rst (48)
> A vision.rst (108)
> 
> -- Patch Links --
> 
>  https://github.com/astropy/astropy-website/pull/1.patch
>  https://github.com/astropy/astropy-website/pull/1.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/astropy/astropy-website/pull/1

embray commented on _themes/bootstrap/layout.html:
```diff
@@ -0,0 +1,119 @@
+        <ul class="nav secondary-nav">
+          {% block sidebarsearch %}
+            {% include "searchbox.html" %}
+          {% endblock %}
```
&lt;/li&gt;

astrofrog committed:
Change DOCTYPE to HTML5

astrofrog committed:
Added missing <li></li>

embray commented:
The last bug is in the base Sphinx theme itself, but it's a really really minor one and I'm not going to worry about it (rel="top" is dropped in HTML5; it should be rel="start" but whatever).

embray commented:
@taldcroft Good suggestion! My one concern is that on the Home page it would lead to logo overkill.  So maybe it should only appear in the top banner on non-home pages.  That should be doable though.

astrofrog committed:
Minor fix

astrofrog commented:
@iguananaut - I've fixed all the issues you reported (except for the `rel='start'`, but since I think that comes from Sphinx itself, not worth worrying about as you said)

@taldcroft - I also like your suggestion. I thought maybe we could have the logo fully transparent (no white) in small instead of 'Home'. I tried briefly, but I couldn't get it to work quite like I wanted. Maybe we can delay that to another pull request, and if anyone wants to give it a shot I can send them the logo?

astrofrog commented:
@taldcroft - regarding the 'fork me on github', what browser/operating system are you using?

taldcroft commented on installation.rst:
```diff
@@ -0,0 +1,47 @@
+You can check that astropy is correctly installed by starting up ``python`` or ``ipython``, and importing ``astropy``::
+
+    >>> import astropy
+
```
Should we suggest doing `>>> astropy.test()` at this point?

astrofrog replied:
That crossed my mind too, and I was trying to think whether there were any reasons not to? (cc @eteq)

eteq replied:
This seems fine, although you might want to make it a little clearer that it's optional, and/or that minor failures aren't necessarily big problems.  

Basically, I just want to make sure users don't run the tests, see some small platform-specific error in a package they aren't going to use, and throw up their hands saying "These Astropy people are totally incompetent!"

taldcroft commented:
@astrofrog About the github fork, I'm using chrome 18.0.1025.151 beta on snow leopard.  What's interesting is that I went to the page source and clicked on the image link, and this sent me to a Github page with an error that "the page did not respond fast enough".  That's probably what was happening in my original view of the astropy docs.  But I clicked on the image link again and this time it worked, and now the astropy page is showing the github fork logo OK (I guess it got cached along the way).

Is there a reason for pulling the image from github assets instead of having it local?

astrofrog committed:
Added 'Fork me on Github' image to the repository

astrofrog commented:
@taldcroft - good point, I don't see a reason not to have it locally, so I've changed that. Can you confirm it works fine?

taldcroft commented:
@astrofrog - The local fork_me works for me.

kbarbary commented:
@taldcroft - I agree about the "@ AstroPy" (or "@ astropy") being on the top bar in place of "Home". Think its fine to omit "Home": I feel like people expect that if you click on a site logo you're taken to the homepage. Some hover text could tell you this as well.

@iguananaut - Also agree it should definitely not be duplicated on the homepage, but for consistency perhaps we just shouldn't have the big version on the homepage. Instead, the tagline could be the title of a paragraph or box giving the elevator pitch.

taldcroft commented:
@kbarbary @iguananaut - The full logo (snake spiral + stylized astropy) is fantastic and it would be a crime to not have it on the home page at full size!  

I also really like the "A Community Python Library for Astronomy" tagline.

kbarbary commented:
@taldcroft - Thanks! Another option would be to make the top bar taller to allow for a bigger logo (but keep the nav items the current size and bottom aligned, if you can imagine what I mean).

For the 'fork me on github' image, should it only be on the relevant page(s), like 'Contributing'? (if this is easy) A lot of eventual visitors to the page will not need to know anything about github.

astrofrog commented:
Ok, I tried with the @ astropy in the menu bar, but actually found that this looks better:

![Screensho](http://www.mpia-hd.mpg.de/~robitaille/share/astropy_preview.png)

If people are happy with this, I can try and implement it properly.

kbarbary commented:
Personally, I'm not a fan of the banner on the homepage, first because it is repetitive with the topbar, and second because it looks like it is floating in the middle of nowhere (perhaps this is because I'm used to seeing it above the navbar). @astrofrog you didn't like something more like this...?

![](http://supernova.lbl.gov/~kbarbary/share/astropy-topnavbar-example.png)

... but of course with the css fixed to not overlap the content area...

astrofrog commented:
@kbarbary - I prefer what I was suggesting, because I think it is nice to have the full logo on the front page (and I don't like the idea of the menubar changing depending on whether we are on the front page or other pages). I don't think it looks weird to have the logo floating under the menu. I do think we want to put part of the logo at least in the menubar, but I found that it looked better without the galaxy part (as I showed above). The galaxy icon will already be in the address bar in most browsers (because of the favicon), so I don't think it's a big issue. What do other people think (@iguananaut and @taldcroft?)

kbarbary commented:
@astrofrog - To be clear, I am suggesting that this always be the menubar on every page, not just the front page. But fair enough to your other comments. Let's see what others think...

taldcroft commented:
I'm with @astrofrog on this one.  His attached screenshot looks really good and I don't find anything awkward with the homepage banner and its placement.

I agree with @kbarbary that we should consider having the fork me image only on the contributing page if that's possible.  (Maybe just put as an issue for now).

astrofrog committed:
Show 'Fork me on Github' only on Contributing page

astrofrog committed:
Include partial logo in navigation bar

astrofrog committed:
Added running of astropy.test() to 'Testing your installation'

astrofrog commented:
I've updated the pages according to the comments (though we can still change which logo goes where if needed)

The 'Fork me on Github' banner will now only appear on the Contributing page

I also added `astropy.test()` to 'Testing your installation'. However, what happens if there is one error/fail in the test - should we tell people they can still use the library, but to report the issue?

astrofrog commented:
(none of the changes are set in stone, so I can always revert some of the commits if needed)

kbarbary commented:
Alright... I can get on board with this :)

On astrofrog.github.com/astropy-website/installation.html (and the other non-home pages) I'm seeing the word "Home" under the logo text in the navbar.

Nitpicking: In the navbar logo, can you reduce the margin between the logo text and the next menu item? Right now it is something like ~25px on the left and ~45 on the right. I think making it similar to the other menu items and the top and bottom borders (~15px) would be better. Or at least left/right symmetric. I suppose there's some reason you implemented this as a background image in the css rather than just an `<img>` in the html?

embray commented:
Agreed with @kbarbary's comments about margins.  As for not using an img tag: That's common practice for any "image" appearing on a site that's just part of the layout, and is not relevant to the content. Though that reminds me, we should add whatever header link/meta tags are appropriate to point to the Astropy logo as the preview image to use when linking to the site on social networking sites.

Should also at some point add a link for the Astropy Twitter...not that that's doing anything right now.

But wow, this looks great! I love the current iteration.

astrofrog committed:
Tweaks to logo in navigation bar

astrofrog commented:
@iguananaut @kbarbary - I fixed the margins for the navigation bar logo!

astrofrog committed:
Added link to Twitter and user mailing list

astrofrog commented:
I added a simple link to twitter and to the user mailing list on the front page for users to stay updated. When we have time, we can figure out if there are more fun ways to link to twitter. We might want to add ShareThis (or similar) buttons, and Google Analytics tracking.

eteq commented:
Regarding the banner/logo placement: I personally like having the logo itself appear in the banner - although I would say a scaled-down version (sized to match the text that's there now) might be better than the larger version in @kbarbary's screenshot above.  Not all browsers do a great job of rendering favicons, and it's such a pretty logo! :)  But I've never been that much of a web design person, so I'll defer to the experience of others if this is the consensus.

@astrofrog - what do you think about also having the "Fork me on Github" banner on the front page (along with the contributing page)?  One of the things I most like about that banner is that I can just go to "astropy.org" and then click on one link to go straight to the github page - and somehow having to do the second click on "contributing" makes it seem a bit less slick (although that's admittedly very subjective).

Also, @astrofrog - I like the idea of adding various share buttons (including google +1, perhaps facebook like, perhaps a share on facebook, etc.) and Analytics... but those might be better as later PRs (although I think that's what you had in mind?)  Perhaps add issues for it after this is merged.

astrofrog commented:
@eteq - actually the fork me banner used to be on all pages, but @kbarbary and @taldcroft suggested I have it only on contributing. I personally prefer it on all pages.

I'll try and make a version with the small logo in the menu with the height to match the text, and we can then compare. Whatever we do, I strongly feel we should have the full sized logo on the front page though.

eteq commented:
@astrofrog  - I absolutely agree the big logo should be on the front page - I'll echo everyone else's comments that it looks just beautiful!

As for the fork me banner, I was trying for compromise by saying just the front and "contributing" :)  But perhaps this is just a matter of conflicting personal preferences...

astrofrog commented:
All right, here's what the front page would look like if I add the galaxy part of the logo to the navigation bar:

![front page](http://www.mpia.de/~robitaille/share/front.png)

and the installation page for example:

![front page](http://www.mpia.de/~robitaille/share/install.png)

You can compare this to the 'live' version above, which is the same but without the logo in the navigation bar.

Thoughts?

astrofrog committed:
Minor fixes

eteq commented:
@astrofrog - I like the version with the small logo better, but perhaps that's obvious from my earlier comment...

kbarbary commented:
I like the version with the small logo better on the non-homepages, but I'm worried about logo overkill on the homepage. (especially for browsers with the favicon in the address bar and the tab!) So... I could go either way.

astrofrog commented:
@taldcroft and @iguananaut - what do you think of the version with the small logo shown in my comment above vs the 'published' version? I see the value in the small logo for the pages that are not the front page, and I don't think it clashes that much with the big one on the front page, but interested to hear other opinions.

I'll plan on updating the source with the small logo version and merging tomorrow if there are no further comments.

astrofrog commented:
Just to clarify my position, I do think the 'published' version looks better, with only the word-part of the logo in the menubar, but wondering whether it is worth the slight downside of having the full logo in the nav bar for front page vs the upside of having it for other pages (and I don't like the idea of changing the navbar icon depending on the page one is on). I'll sleep over it...

kbarbary commented:
Side note: I don't think its necessary to have the logo credit in the page footer. We can just have a "credit" file that lives with the source.

eteq commented:
@kbarbary - FYI, that's already present in the main astropy docs (see `astropy/docs/logo` in astropy/astropy@55dd431f)

embray commented:
Sorry, just getting back to this. I think I prefer without the logo in the nav bar.  But I don't think it's horrible either.  So I'll punt and say "-0".

I think also like to have the "Fork Me" banner on the front index, but I won't cry if it's not there either :)

astrofrog committed:
Removed logo credit from bottom of pages

astrofrog commented:
All right, I'm going to merge this now, without changing the logo, because I have an idea for a potential solution to this 'issue'. Given that there is consensus anyway apart from this minor point, I'm just merging this, and I'll open a new pull request for suggested modifications.
